esp/services/espcontrol/ws_espcontrolservice.cpp

```diff
@@ -51,3 +131,200 @@ bool CWSESPControlEx::onSetLogging(IEspContext& context, IEspSetLoggingRequest&
+        unsigned port = 8010;
+        if (!req.getPort_isNull())
+            port = req.getPort();
+
+        VStringBuffer xpath("/%s/%s[@name='%s']/%s[@name='%s']/%s[@port=\"%d\"]/%s[%s='%s']", PathSessionRoot, PathSessionProcess, espProcess.get(),
+            PathSessionDomain, authDomain.str(), PathSessionApplication, port, PathSessionSession, PropSessionID, sessionID.str());
+        Owned<IRemoteConnection> globalLock = querySDSConnection(xpath.str(), RTM_LOCK_READ, SESSION_SDS_LOCK_TIMEOUT);
+        setSessionInfo(sessionID.str(), globalLock->queryRoot(), authDomain.str(), port, &resp.updateSession());
+    }
+    catch(IException* e)
+    {
+        FORWARDEXCEPTION(context, e, ECLWATCH_INTERNAL_ERROR);
+    }
+    return true;
+}
+
+bool CWSESPControlEx::onCleanSession(IEspContext& context, IEspCleanSessionRequest& req, IEspCleanSessionResponse& resp)
+{
+    try
+    {
+#ifdef _USE_OPENLDAP
+        CLdapSecManager* secmgr = dynamic_cast<CLdapSecManager*>(context.querySecManager());
+        if(secmgr && !secmgr->isSuperUser(context.queryUser()))
+            throw MakeStringException(ECLWATCH_SUPER_USER_ACCESS_DENIED, "Failed to clean session. Permission denied.");
+#endif
+
+        StringBuffer sessionID = req.getSessionID();
+        StringBuffer fromIP = req.getFromIP();
+        if ((sessionID.trim().isEmpty()) && (fromIP.trim().isEmpty()))
+            throw MakeStringException(ECLWATCH_INVALID_INPUT, "SessionID or FromIP has to be specified.");
+
+        StringBuffer authDomain = req.getAuthDomain();
+        if (authDomain.trim().isEmpty())
+            authDomain.set("default");
+        unsigned port = 8010;
+        if (!req.getPort_isNull())
+            port = req.getPort();
+
+        VStringBuffer xpath("/%s/%s[@name='%s']/%s[@name='%s']/%s[@port=\"%d\"]", PathSessionRoot, PathSessionProcess,
+            espProcess.get(), PathSessionDomain, authDomain.str(), PathSessionApplication, port);
+        Owned<IRemoteConnection> globalLock = querySDSConnection(xpath.str(), RTM_LOCK_WRITE, SESSION_SDS_LOCK_TIMEOUT);
+
+        IArrayOf<IPropertyTree> toRemove;
+        if (sessionID && *sessionID)
+            xpath.setf("%s[%s='%s']", PathSessionSession, PropSessionID, sessionID.str());
+        else
+            xpath.setf("%s[%s='%s']", PathSessionSession, PropSessionNetworkAddress, fromIP.str());
+        Owned<IPropertyTreeIterator> iter = globalLock->queryRoot()->getElements(xpath.str());
+        ForEach(*iter)
+            toRemove.append(*LINK(&iter->query()));
+        ForEachItemIn(i, toRemove)
+            globalLock->queryRoot()->removeTree(&toRemove.item(i));
+
+        resp.setStatus(0);
+        resp.setMessage("Session is cleaned.");
+    }
+    catch(IException* e)
+    {
+        FORWARDEXCEPTION(context, e, ECLWATCH_INTERNAL_ERROR);
+    }
+    return true;
+}
+
+bool CWSESPControlEx::onSetSessionTimeout(IEspContext& context, IEspSetSessionTimeoutRequest& req, IEspSetSessionTimeoutResponse& resp)
+{
```

Comment:
Only 'super user' can access those methods. Is that not good enough? Do we need to create a new user category: 'super super user'?